src/jit/importer.cpp

```diff
@@ -3865,9 +3870,392 @@ NamedIntrinsic Compiler::lookupNamedIntrinsic(CORINFO_METHOD_HANDLE method)
         }
     }
 
+#ifdef _TARGET_XARCH_
+    if ((namespaceName != nullptr) && strcmp(namespaceName, "System.Runtime.Intrinsics.X86") == 0)
+    {
+        InstructionSet isa = lookupHWIntrinsicISA(className);
+        result             = lookupHWIntrinsic(methodName, isa);
+    }
+#endif
     return result;
 }
 
+#ifdef _TARGET_XARCH_
```

Comment:
Could all this new code be moved to a separate file? `importerxarch.cpp` perhaps? This file is already large as is.

Reply:
Yes, I will separate this new code from importer.cpp to a new file. How about `hwintrinsicxarch.pp`?

Reply:
> How about hwintrinsicxarch.pp?

I assume you mean .cpp. Sounds reasonable to me.

Reply:
Yes 😄